### PR TITLE
Add --second-docker-image option

### PR DESCRIPTION
**Pull request by tonyjohnchen**
## Fixes / Features
New feature:
Add `--second-docker-image` option to allow user provide a second docker image to merge with `base_docker_image`.
The new merged docker image will be used as `base_docker_image`.
This option is helpful if you want to run two features from to docker image within one XPK workload. In my case, it was `Maxtext AOT` + `Hybridsim`

## Testing / Documentation
Test command:
```
export MAXTEXT_DOCKER_IMAGE=gcr.io/tpu-prod-env-multipod/maxtext_jax_nightly:latest
export HYBRIDSIM_DOCKER_IMAGE=gcr.io/tpu-prod-env-multipod/hybridsim:prod-latest

python3 xpk.py workload create \
--workload tony-aot-hybridsim-prod-test \
--base-docker-image=$MAXTEXT_DOCKER_IMAGE \
--second-docker-image=$HYBRIDSIM_DOCKER_IMAGE \
--command "gsutil cp gs://tonyjohnchen-hybridsim/hybridsim-prod/run_hybridsim.sh ./ && bash run_hybridsim.sh" \
--cluster v4-8-maxtext \
--num-slices=1 \
--tpu-type=v4-8
```

Test Logs:
http://shortn/_4vPlVMJb7u

- [ y ] Tests pass
- [ y/n ] Appropriate changes to documentation are included in the PR

### CODE REVIEW

File: xpk.py [hunk 1]
```diff
@@ -4842,6 +4869,15 @@ def directory_path_type(value):
         ' will be added to the image.'
     ),
 )
+workload_base_docker_image_arguments.add_argument(
```
Comment (Obliviour) [inline xpk.py:4872]: What about `--docker-image` flow? Can we support adding a second image from it?

File: xpk.py [hunk 2]
```diff
@@ -443,6 +443,20 @@
 WORKDIR /app
```
Comment (Obliviour) [inline xpk.py:443]: Add a readme detail about this flag once we finalized how it looks!

**Review by Obliviour — COMMENTED** [review]
Added some comments, lets talk more about the HybridSim use case of this flag first. Thanks!

**Comment by tonyjohnchen** [comment]
> I assume this PR is not needed anymore? @tonyjohnchen let me know if we can close it!

Yeah, I can close it.

**Pull request closed without merging**